### PR TITLE
chore: handle search result empty types after graphql-codegen upgrade

### PR DESCRIPTION
**Pull request by krzysu**
context discussion https://github.com/dotansimha/graphql-code-generator/discussions/5567

### CODE REVIEW

File: packages/client/src/search/Search.ts [hunk 1]
```diff
@@ -21,19 +21,26 @@ export class Search {
...
 
   async publications(
     request: SearchPublicationsQueryVariables,
   ): Promise<PaginatedResult<CommentFragment | PostFragment>> {
     return buildPaginatedQueryResult(async (variables) => {
-      const result = await this.sdk.SearchPublications(variables);
+      const response = await this.sdk.SearchPublications(variables);
+      const result = response.data.result;
+
+      if (result.__typename === 'ProfileSearchResult') throw new Error('This should never happen');
```
Comment (krzysu) [inline packages/client/src/search/Search.ts:41]: what do you think about a solution like this?

File: packages/client/src/search/Search.ts [hunk 2]
```diff
@@ -21,19 +22,29 @@ export class Search {
...
   async publications(
     request: SearchPublicationsQueryVariables,
   ): Promise<PaginatedResult<CommentFragment | PostFragment>> {
     return buildPaginatedQueryResult(async (variables) => {
-      const result = await this.sdk.SearchPublications(variables);
+      const response = await this.sdk.SearchPublications(variables);
+      const result = response.data.result;
+
+      if (result.__typename === 'ProfileSearchResult') {
+        never('ProfileSearchResult are not expected in this query');
```
Comment (krzysu) [inline packages/client/src/search/Search.ts:44]: what do you think about a solution like this?

File: packages/client/src/helpers/buildPaginatedQueryResult.spec.ts [hunk 3]
```diff
@@ -1,33 +1,37 @@
...
 
 describe('Given a paginated query function and the paginated query result helper', () => {
   const sdk = getSdk(new GraphQLClient(mumbaiSandbox.gqlEndpoint));
 
   it('should be able to paginate forward', async () => {
     const result = await buildPaginatedQueryResult(
-      async (variables) => {
-        const res = await sdk.SearchProfiles(variables);
+      async (currRequest) => {
+        const res = await sdk.Profiles({
```
Comment (krzysu) [inline packages/client/src/helpers/buildPaginatedQueryResult.spec.ts:13]: I changed to less "troublesome" query in this test

File: packages/client/src/search/Search.ts (hunk 2 above)
Comment (desfero) [reply packages/client/src/search/Search.ts:44]: looks good, we can also go with `invariant(result.__typename === "PublicationSearchResult", ...)` to have just one line but that's nit

**Review by desfero — APPROVED** [review]

**Pull request merged**